plugins/debug-console.js

```diff
@@ -112,80 +118,44 @@ function overwriteNative() {
   overwrite('debug');
 }
 
-function setupPosting() {
-  if (!window.isSmartphone()) {
-    $('#chatinput input').keydown(function(event) {
-      var kc = event.keyCode ? event.keyCode : event.which;
-      if (kc === 13) { // enter
-        if ($('#chatcontrols .active').text() === 'debug') {
-          event.preventDefault();
-          userInput();
-        }
-      }
-    });
-  }
-
-  $('#chatinput').submit(function(event) {
-    event.preventDefault();
-    userInput();
-  });
-}
-
-function userInput() {
-  if ($('#chatcontrols .active').text() !== 'debug') {return;}
-
-  var msg = $.trim($('#chatinput input').val());
-  if (!msg) { return; }
-
-  var result;
-  try {
-    result = eval(msg);
-  } catch (e) {
-    if (e.stack) { debugTab.console.error(e.stack); }
-    throw e; // to trigger native error message
-  }
-  if (result !== undefined) {
-    debugTab.console.log(result.toString());
-  }
-}
-
-
-function create() {
-  if ($('#debugconsole').length) return;
-  $('#chatcontrols').append('<a>debug</a>');
-  $('#chatcontrols a:last').click(debugTab.console.show);
-  $('#chat').append('<div style="display: none" id="debugconsole"><table></table></div>');
+// Old API utils
+debugTab.renderDetails = function () {
+  debugTab.console.log('portals: ' + Object.keys(window.portals).length);
+  debugTab.console.log('links:   ' + Object.keys(window.links).length);
+  debugTab.console.log('fields:  ' + Object.keys(window.fields).length);
+};
 
-  setupPosting();
+debugTab.printStackTrace = function () {
+  var e = new Error('dummy');
+  debugTab.console.error(e.stack);
+  return e.stack;
+};
 
-  if (window.useAndroidPanes()) {
-    android.addPane('debug', 'Debug', 'ic_action_view_as_list');
-    window.addHook('paneChanged', function (id) {
-      if (id === 'debug') {
-        debugTab.console.show();
-      }
-    });
-  }
-}
+debugTab.show = function () {
+  window.chat.show('debug');
+};
 
 function setup() {
   window.plugin.debug = debugTab;
-  create();
+  debugTab.create();
   overwriteNative();
 
+  $('<style>')
+    .text('#chat #chatdebug td:nth-child(-n+2) { \n  width: 51px\n' + '}\n#chat #chatdebug td:nth-child(3) {\n  font-family: monospace\n}')
```

Comment:
Styles in separate CSS with import/include here?

Reply:
I only intended to adapt debug-console but that's a good occasion to improve it a bit :+1: